packages/features/data-table/components/DataTablePagination.tsx

```diff
@@ -36,10 +36,9 @@ export function DataTablePagination<TData>({
         currentPage={pageIndex + 1}
         pageSize={pageSize}
         totalItems={totalRowCount}
-        onPageChange={(page) => setPageIndex(page - 1)}
         onPageSizeChange={(newSize) => setPageSize(newSize)}
         onNext={() => setPageIndex(pageIndex + 1)}
-        onPrevious={() => setPageIndex(pageIndex - 1)}
+        onPrevious={() => setPageIndex(pageIndex - 1 == 0 ? null : pageIndex - 1)}
```

Comment:
If we provide a setPageIndex wrapper function in the provider, we won't need this change! (easier on the consumer's side)